### ISSUE TITLE
Include ffmpeg.wasm as a renderer drop-in

### ISSUE THREAD

**Issue by tonyketcham** [genbs/urpflanze#8]
**Is your feature request related to a problem? Please describe.**

Including additional client-side renderer examples in the docs would give a greater wealth of modular options as far as renderers go, since you're free to drop in whatever you want to this stage of the pipeline.

**Describe the solution you'd like**

[FFmpeg.wasm](https://ffmpegwasm.github.io/) packages a bunch of options for output animations/video able to be rendered in the browser client-side, or on a Node server. It can render a frame buffer as gifs, WebM, WebP, mp4/H.264, H.256, etc, and supports multi-threading for mp4 rendering. The library also supports audio encoding which may come in handy for the future plans of building in an audio pipeline. 

Adding an example using FFmpeg.wasm to the docs would be awesome! Probably no need to bundle the renderer directly in _urpflanze_ since it's quite modular by design. This could be another solid alternative to using the raw zip export or [mattdesl/mp4-h264](https://github.com/mattdesl/mp4-h264).

**Comment by genbs** [comment]
You caught a very interesting point.
I didn't give much explanation on rendering as I knew my solution was only momentary.

The animations in Urpflanze work based on time, I have adopted this solution instead of the deltaTime so that I can manage
better animation loops.
Based on this decision I have implemented the [Timeline](https://github.com/genbs/urpflanze/blob/master/src/services/timeline/Timeline.ts) which deals with managing the time and the framerate.

The update flow happens like this:
requestAnimationFrame (get the currentTime) -> [timeline tick](https://github.com/genbs/urpflanze/blob/master/src/services/timeline/Timeline.ts#L199) (set the time and the current frame) -> update the scene in the current time -> drawing
For now update and drawing are [performed linearly](https://github.com/genbs/urpflanze/blob/master/src/services/drawers/drawer-canvas/DrawerCanvas.ts#L362)

The current Renderer works like this:
I scroll the frames from 0 to count -> set the current frame in the timeline and calculate the time -> drawing -> store the Uint8Array of the current frame
I needed the zip to put the images in a package, but since I had memory problems, I divided it into parts.


As you said, the rendering part could be an external service, in fact initially all those that are present in the services
they could be isolated modules (@urpflanze/svg-importer, @urpflanze/drawer-webgl, @urpflanze/synth, @urpflanze/gui), but for a matter of ease of use I preferred to release it in a single package, also because I don't know if it will be used.

Another idea was to create something with glsl as a postprocessing effect (I wrote down a few lines of code but I want to study the integration with webgl better)

but I would leave the option to export frames as pngs since I was thinking about using shapes as moving masks in editing software (haven't tried yet).
